craft_application/commands/remote.py

```diff
@@ -0,0 +1,206 @@
+--status option.
+
+To set a timeout on the remote-build command, use the option
+``--launchpad-timeout=<seconds>``. The timeout is local, so
+the build on launchpad will continue even if the local instance
+is interrupted or times out.
+"""
+
+_CONFIRMATION_PROMPT = (
+    "All data sent to remote builders will be publicly available. "
+    "Are you sure you want to continue?"
+)
+
+
+class RemoteBuild(ExtensibleCommand):
+    """Build a project on Launchpad."""
+
+    name = "remote-build"
+    help_msg = "Build a project remotely on Launchpad."
+    overview = OVERVIEW
+    always_load_project = True
+
+    @override
+    def _fill_parser(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "--recover", action="store_true", help="recover an interrupted build"
+        )
+        parser.add_argument(
+            "--launchpad-accept-public-upload",
+            action="store_true",
+            help="acknowledge that uploaded code will be publicly available.",
+        )
+        parser.add_argument(
+            "--launchpad-timeout",
+            type=int,
+            default=0,
+            metavar="<seconds>",
+            help="Time in seconds to wait for launchpad to build.",
+        )
+
+    def _run(
+        self,
+        parsed_args: argparse.Namespace,
+        **_kwargs: Any,  # noqa: ANN401 (use of Any)
+    ) -> int | None:
+        """Run the remote-build command.
+
+        :param parsed_args: parsed argument namespace from craft_cli.
+
+        :raises AcceptPublicUploadError: If the user does not agree to upload data.
+        """
+        if os.getenv("SUDO_USER") and os.geteuid() == 0:
+            emit.progress(
+                "Running with 'sudo' may cause permission errors and is discouraged.",
+                permanent=True,
+            )
+            # Give the user a bit of time to process this before proceeding.
+            time.sleep(1)
+
+        emit.progress(
+            "remote-build is experimental and is subject to change. Use with caution.",
+            permanent=True,
+        )
+
+        if (
+            not parsed_args.launchpad_accept_public_upload
+            and not util.confirm_with_user(_CONFIRMATION_PROMPT, default=False)
+        ):
+            emit.message("Cannot proceed without accepting a public upload.")
+            return 77  # permission denied from sysexits.h
+
+        builder = self._services.remote_build
+        project = cast(models.Project, self._services.project)
+        config = cast(dict[str, Any], self.config)
+        project_dir = (
+            pathlib.Path(config.get("global_args", {}).get("project_dir") or ".")
+            .expanduser()
+            .resolve()
+        )
+        emit.trace(f"Project directory: {project_dir}")
+
+        if parsed_args.launchpad_timeout:
+            emit.debug(f"Setting timeout to {parsed_args.launchpad_timeout} seconds")
+            builder.set_timeout(parsed_args.launchpad_timeout)
```

Comment:
Side question:
Is `codecov` not running on the PR's that originates from the same repo :thinking: ? 
I can see a lot of branches here that are not covered, but there are also no complaints from coverage

Reply:
very true, I don't know what's going on
I have vague recollections that the codecov service stopped working?

Reply:
I know it doesn't have the latest `main`, but it triggers fine on the PR's that originate from forks.
https://github.com/canonical/craft-application/pull/587